src/components/TodoContainer.jsx

```diff
@@ -0,0 +1,132 @@
+            }));
+
+            // Updating state with the fetched todos
+            setTodoList(todos);
+            setIsLoading(false);
+        } catch (error) {
+            console.error(error.message);
+        }
+    };
+
+    // hook to fetch data from API
+    useEffect(() => {
+        fetchData();
+    }, []);
+
+    // posting a new todo to the list
+    const addTodo = async (title) => {
+        const titleData = {
+            fields: {
+                title: title,
+            },
+        };
+        const url = `https://api.airtable.com/v0/${baseId}/${tableName}`;
+        const options = {
+            method: "POST",
+            headers: {
+                "Content-Type": "application/json",
+                Authorization: `Bearer ${apiKey}`,
+            },
+            body: JSON.stringify(titleData),
+        };
+        try {
+            console.log('Request Payload:', JSON.stringify(titleData, null, 2));
+            const response = await fetch(url, options);
+            if (!response.ok) {
+                throw new Error(`Error has occurred: ${response.status}`);
+            }
+            const todo = await response.json();
+            const newTodo = { id: todo.id, title: todo.fields.title };
+            setTodoList([...todoList, newTodo]);
+        } catch (error) {
+            console.log(error.message);
+            return null;
+        }
+    };
+
+    // removing a todo from the list by its id from remote site
+    const removeTodo = async (id) => {
+        const url = `https://api.airtable.com/v0/${baseId}/${tableName}/${id}`;
+        const options = {
+            method: "DELETE",
+            headers: {
+                "Content-Type": "application/json",
+                Authorization: `Bearer ${apiKey}`,
+            },
+        };
+        try {
+            const response = await fetch(url, options);
+            if (!response.ok) {
+                throw new Error(`Error has occurred: ${response.status}`);
+            }
+            const newTodoList = todoList.filter(function (todo) {
+                return id !== todo.id;
+            });
+            setTodoList(newTodoList);
+        } catch (error) {
+            console.log(error.message);
+        }
+    };
+
+    return (
+        <>
+            <div style={{display: 'flex', alignItems: 'center', justifyContent: 'center'}}>
+                <img src={listIcon} alt="List Icon"
+                     style={{width: '48px', height: '48px', marginRight: '16px'}}/>
+                <h1>Todo List</h1>
+            </div>
+            {isLoading ? (
+                <p className="Loading">Loading...</p>
+            ) : (
+                <>
+                    <AddTodoForm onAddTodo={addTodo} todoList={todoList}/>
+                    <TodoList todoList={todoList} onRemoveTodo={removeTodo}/>
+                </>
+            )}
+        </>
+    )
+}
+
+TodoContainer.propTypes = {
+    tableName: PropTypes.string.isRequired,
```

Comment:
just interesting: why only in this component you have a flag: isRequires? Does it mean in other components the props are optional?

Reply:
When building components for applications that interact with APIs, using .isRequired for props is indeed super important.